netsuite/vendor.bal

```diff
@@ -0,0 +1,261 @@
+        legalName: extractStringFromXML(response/**/<listRel:legalName>/*),
+        vatRegNumber: extractStringFromXML(response/**/<listRel:vatRegNumber>/*),
+        expenseAccount: extractRecordRefFromXML(response/**/<listRel:expenseAccount>),
+        payablesAccount: extractRecordRefFromXML(response/**/<listRel:payablesAccount>),
+        terms: extractRecordRefFromXML(response/**/<listRel:terms>),
+        incoterm: extractRecordRefFromXML(response/**/<listRel:incoterm>),
+        creditLimit: extractDecimalFromXML(response/**/<listRel:creditLimit>/*),
+        balancePrimary: extractDecimalFromXML(response/**/<listRel:balancePrimary>/*),
+        openingBalance: extractDecimalFromXML(response/**/<listRel:openingBalance>/*),
+        openingBalanceDate: extractStringFromXML(response/**/<listRel:openingBalanceDate>/*),
+        openingBalanceAccount: extractRecordRefFromXML(response/**/<listRel:openingBalanceAccount>),
+        balance: extractDecimalFromXML(response/**/<listRel:balance>/*),
+        unbilledOrdersPrimary: extractDecimalFromXML(response/**/<listRel:unbilledOrdersPrimary>/*),
+        bcn: extractStringFromXML(response/**/<listRel:bcn>/*),
+        unbilledOrders: extractDecimalFromXML(response/**/<listRel:unbilledOrders>/*),
+        currency: extractRecordRefFromXML(response/**/<listRel:currency>),
+        laborCost: extractDecimalFromXML(response/**/<listRel:laborCost>/*),
+        purchaseOrderQuantity: extractDecimalFromXML(response/**/<listRel:purchaseOrderQuantity>/*),
+        purchaseOrderAmount: extractDecimalFromXML(response/**/<listRel:purchaseOrderAmount>/*),
+        purchaseOrderQuantityDiff: extractDecimalFromXML(response/**/<listRel:purchaseOrderQuantityDiff>/*),
+        receiptQuantity: extractDecimalFromXML(response/**/<listRel:receiptQuantity>/*),
+        receiptAmount: extractDecimalFromXML(response/**/<listRel:receiptAmount>/*),
+        receiptQuantityDiff: extractDecimalFromXML(response/**/<listRel:receiptQuantityDiff>/*),
+        workCalendar: extractRecordRefFromXML(response/**/<listRel:workCalendar>),
+        taxIdNum: extractStringFromXML(response/**/<listRel:taxIdNum>/*),
+        taxItem: extractRecordRefFromXML(response/**/<listRel:taxItem>),
+        password: extractStringFromXML(response/**/<listRel:password>/*),
+        password2: extractStringFromXML(response/**/<listRel:password2>/*),
+        defaultTaxReg: extractRecordRefFromXML(response/**/<listRel:defaultTaxReg>),
+        predictedDays: extractIntegerFromXML(response/**/<listRel:predictedDays>),
+        predConfidence: extractDecimalFromXML(response/**/<listRel:predConfidence>/*)
+    };
+
+    var customFields = extractCustomFiledListFromXML(response/**/<listRel:customFieldList>/*);
+    if (customFields is CustomFieldList) {
+        vendor.customFieldList = customFields;
+    }
+
+    boolean|error value = extractBooleanValueFromXMLOrText(response/**/<listRel:isInactive>/*);
```

Comment:
Cant we do like
`boolean value = check extractBooleanValueFromXMLOrText(response/**/<listRel:isInactive>/*);`

So we can get rid of all the `if` blocks below this

Reply:
This is used in NetSuite xml result to ballerina record manual mapping as still XML to ballerina record mapping is not fully supported. NetSuite does not provide values for all the fields in the record for every request. There can be missing XML elements. it depends on the business logic and the values of the NetSuite record. Therefore some records might have values for some fields, but they might not be in another record. If we use 'check' we will get an error every time when Netsuite doesn't provide values for all fields. As we use manual mapping with xml payloads, we can't get rid of if blocks. But when we got full XML to record mapping support, these all can be removed.

Reply:
+1